### ISSUE TITLE
Login id input does not have correct type attribute

### ISSUE THREAD

**Issue by elliotdickison** [FusionAuth/fusionauth-issues#1556]
### Description 

When "login type" is set to "email", the loginId input (rendered by the "OAuth register", "OAuth authorize", and "Forgot password" templates from the default theme) has a `type="text"` attribute instead of `type="email"`. This is mostly a problem on mobile - instead of the device showing an email keyboard (with keys like "@" and ".com" readily available) it shows the standard keyboard (where you have to hunt for those options).

### Affects versions

Tested with 1.30.1

### Steps to reproduce

- Setup an application that uses the default FusionAuth theme and has login type set to "email".
- Trigger an OAuth registration, OAuth authorization, or Forgot password flow for the application
- Inspect the email input and note that the input's type is "text"

### Expected behavior

I would expect setting "login type" to "email" to instruct the default theme to render an email input for the login id.

### Platform

All

### Community guidelines
All issues filed in this repository must abide by the [FusionAuth community guidelines](https://fusionauth.io/community/forum/topic/1000/code-of-conduct).

### Additional context

I realize we could fix this pretty easily ourselves with some tweaks to the custom theme (which we'll probably do), but it would be nice if this behavior was built-in. We try to be careful with theme updates because of the difficulty in upgrading them. We love FusionAuth, keep up the good work!

**Comment by robotdan** [comment]
The `loginId` field can be a username or email address. I don't know if there would be any negative implications to change it to `email` when using a username. 

The preferred option would for you to modify the default theme to change it from `text` to `email` if in your case you know the user's will be entering an email address and it allows for an improved mobile experience.

**Comment by elliotdickison** [comment]
Good idea, we'll do that for now. I just thought it would be nice if the built-in theme automatically adapted based on login type (so "text" when login type is username, "email" when login type is email). Maybe I should have filed under "feature request" instead of "bug report". I think ideally you would expose the login type as a variable to the themes and then update the default theme files to change the input type based on that. Then people building custom themes could apply the same logic to their custom inputs (we are trying to keep all our custom theme work compatible with the options in the FusionAuth UI since it's nice to be able to tweak settings there).

**Comment by robotdan** [comment]
That is an option. Everything we build uses our own API, including our login pages. Currently the Login API takes a `loginId` input parameter which can be either a `username` or `email`. 

We are working towards making identify types extendable, so we may have a phone number, email, username, govt. id, etc. as an identify. When we get there.. perhaps we should revisit this and allow the system to be configured to only allow a particular type of identity at the API level. If we know for certain that we would only allow a particular identify type (email, username, etc) - then we could certainly be smarter when we build the input field.
